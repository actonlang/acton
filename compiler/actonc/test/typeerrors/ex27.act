class Foo(object):
    def __init__(self):
        pass

    def foo(a: int):
        return 42
