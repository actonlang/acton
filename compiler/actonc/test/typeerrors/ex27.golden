Building file test/typeerrors/ex27.act using temporary scratch directory
  Compiling ex27.act for development
[error]: Type unification error
     +--> ex27.act@5:5-6:1
     |
   5 | +>     def foo(a: int):
   6 | |>         return 42
     : | 
     : `- Type of first parameter of class method does not unify with Self
-----+
